src/Microsoft.DotNet.Configurer/DotnetFirstRunConfiguration.cs

```diff
@@ -0,0 +1,14 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.DotNet.Configurer
+{
+    public class DotnetFirstRunConfiguration
+    {
+        public bool GenerateAspNetCertificate { get; set; }
```

Comment:
I don't think this class can help too much. There is no behavior moved to this this class, so it is yet another flag.

It is still a good thing if we can move all the fields to one object. Although, why there is no other fields like expending cache or not. And this is a input(user intention) or output(end state)?

Reply:
Looks like it is a straight replacement for fields in DotnetFirstTimeUseConfigurer.

DotnetFirstRunConfiguration is too general while it just hold partial state of DotnetFirstTimeUseConfigurer. Maybe make it a subclass of DotnetFirstTimeUseConfigurer?

Reply:
I disagree. I don't believe that the class has only value if it has explicit behavior.

This this case, using a POCO allows us to have a finer control over the values used to control the behavior of the configurer. There is no field for the cache expansion because it is not controlled individually today. The fields here are directly mapping from environment variables to property and this allows us to control those values when we want.

For instance, when running first run through a native installer, I can just set SkipFirstRun to false, independent of the variable value, guaranteeing that first run will always run for native installers.

Reply:
Talked to @livarcocc offline. This is for "end state". Make it an immutable object is good enough